### PR TITLE
xcc first steps

### PR DESCRIPTION
**Pull request by mehtaphysical**
First steps for XCC docs. In the future we'll also need to add:

* testing docs

### CODE REVIEW

**Comment by BenKurrek** [comment]
Much needed tutorial - fantastic job. There are a few things I would add / modify: 
- GAS explanation and how it works/is calculated for cross-contract calls. 
- The following (bolded part) is a little confusing and I didn't quite follow --> "If the cross contract call fails the callback returns oops!. If the cross contract call is successful, the callback returns Wow! **if the accounts balance is > 100000** otherwise it returns Hmmmm."

**Review by bucanero — APPROVED** [review]
LGTM

File: docs/tutorials/contracts/xcc-receipts.md [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+- [Smart Contract Overview](/docs/develop/contracts/overview)
+- [Rust Introduction](/docs/develop/contracts/rust/intro)
+- [AssemblyScript Introduction](/docs/develop/contracts/as/intro)
+
+## Terminology
+
+- **`Runtime`** - the layer responsible for running smart contracts. It converts transactions into receipts and processes receipts. [More info here](https://nomicon.io/RuntimeSpec/Runtime.html)
+- **`Receipt`** - messages passed between shards. [More info here](https://nomicon.io/RuntimeSpec/Receipts.html#receipt)
+- **`ActionReceipt`** - a `Receipt` used to apply some actions to a receiver (such as apply a smart contract method). [More info here](https://nomicon.io/RuntimeSpec/Receipts.html#actionreceipt)
+- **`DataReceipt`** - a `Receipt` used to return. [More info here](https://nomicon.io/RuntimeSpec/Receipts.html#datareceipt)
```
Comment (thisisjoshford) [inline docs/tutorials/contracts/xcc-receipts.md:22]:
```suggestion
- **`DataReceipt`** - represents the final contract execution result . [More info here](https://nomicon.io/RuntimeSpec/Receipts.html#datareceipt)
```

**Comment by bucanero** [comment]
> Much needed tutorial - fantastic job. There are a few things I would add / modify:
>     * The following (bolded part) is a little confusing and I didn't quite follow --> "If the cross contract call fails the callback returns oops!. If the cross contract call is successful, the callback returns Wow! **if the accounts balance is > 100000** otherwise it returns Hmmmm."

I've adjusted the paragraph:
>  If the cross contract call fails the callback returns `oops!`. If the cross contract call is successful and the balance is `> 100000` the callback returns `Wow!`, otherwise it returns `Hmmmm`.

File: docs/tutorials/contracts/xcc-receipts.md [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+## Prerequisites
+
+- [Smart Contract Overview](/docs/develop/contracts/overview)
+- [Rust Introduction](/docs/develop/contracts/rust/intro)
+- [AssemblyScript Introduction](/docs/develop/contracts/as/intro)
+
+## Terminology
+
+- **`Runtime`** - the layer responsible for running smart contracts. It converts transactions into receipts and processes receipts. [More info here](https://nomicon.io/RuntimeSpec/Runtime.html)
+- **`Receipt`** - messages passed between shards. [More info here](https://nomicon.io/RuntimeSpec/Receipts.html#receipt)
```
Comment (thisisjoshford) [inline docs/tutorials/contracts/xcc-receipts.md:20]:
```suggestion
- **`Receipt`** - messages passed between blocks. [More info here](https://nomicon.io/RuntimeSpec/Receipts.html#receipt)
```
```suggestion
- **`Receipt`** - messages passed between shards or on the same shard. [More info here](https://nomicon.io/RuntimeSpec/Receipts.html#receipt)
```

File: docs/tutorials/contracts/xcc-rust-cheatsheet.md [hunk 3]
```diff
@@ -0,0 +1,302 @@
+---
+id: xcc-rust-cheatsheet
+title: Rust Cross Contract Calls Cheat Sheet
+sidebar_label: Rust - Cheat Sheet
+---
+
+In Rust smart contracts, cross contract calls are made by:
+
+1. defining traits
```
Comment (thisisjoshford) [inline docs/tutorials/contracts/xcc-rust-cheatsheet.md:9]:
```suggestion
1. Defining traits
```

File: docs/tutorials/contracts/xcc-rust-cheatsheet.md [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+2. calling methods on the defined traits
```
Comment (thisisjoshford) [inline docs/tutorials/contracts/xcc-rust-cheatsheet.md:10]:
```suggestion
2. Calling methods on the defined traits
```

**Review by thisisjoshford — APPROVED** [review]
Left some minor change suggestions.

**Pull request merged**